node/core/provisioner/src/lib.rs

```diff
@@ -635,7 +652,11 @@ fn extend_by_random_subset_without_repetition(
 async fn select_disputes(
 	sender: &mut impl SubsystemSender,
 	metrics: &metrics::Metrics,
+	disputes_enabled: bool,
 ) -> Result<MultiDisputeStatementSet, Error> {
+	if !disputes_enabled {
```

Comment:
I would move the check outside of `select_disputes`. IMO entering this function implies that disputes are enabled, otherwise why would one pick some ?